### PR TITLE
 (Add Cvar) Flashbang should remain the active item after a throw 

### PR DESCRIPTION
**Pull request by KaidoRen**
Video preview: https://www.youtube.com/watch?v=CtsPT3axwYg

### CODE REVIEW

File: dist/game.cfg [hunk 1]
```diff
@@ -186,3 +186,10 @@ mp_item_staytime 300
 //
 // Default value: "1"
 mp_legacy_bombtarget_touch "1"
+
+// Flashbang should remain the active item after a throw.
+// 0 - disabled
+// 1 - enabled
+//
+// Default value: "0"
+mp_flashbang_activeitem_afterthrow "1"
```
Comment (IgnacioFDM) [inline dist/game.cfg]: Default value "0" but it's 1 there

**Comment by IgnacioFDM** [comment]
Also it'd be nice if this worked for all nades (not just flashbangs), even if in vanilla cs 1.6 you can't have more than 1 of the others.

File: dist/game.cfg (hunk 1 above)
Comment (KaidoRen) [reply dist/game.cfg]: thanks
fixed

**Comment by KaidoRen** [comment]
this problem was only flashbang

**Comment by In-line** [comment]
Previous PR: #258

File: regamedll/dlls/game.cpp [hunk 2]
```diff
@@ -116,6 +116,7 @@ cvar_t show_radioicon          = { "mp_show_radioicon", "1", FCVAR_SERVER, 1.0f,
 cvar_t old_bomb_defused_sound  = { "mp_old_bomb_defused_sound", "1", FCVAR_SERVER, 1.0f, nullptr };
 cvar_t item_staytime           = { "mp_item_staytime", "300", FCVAR_SERVER, 300.0f, nullptr };
 cvar_t legacy_bombtarget_touch = { "mp_legacy_bombtarget_touch", "1", FCVAR_SERVER, 1.0f, nullptr };
+cvar_t flashbang_activeitem_afterthrow = { "mp_flashbang_activeitem_afterthrow", "0", FCVAR_SERVER, 0.0f, nullptr };
```
Comment (In-line) [inline regamedll/dlls/game.cpp]: Maybe something like `mp_flashbang_throw_behavior` or `mp_flashbang_remain_active` is better?

**Comment by In-line** [comment]
@IgnacioFDM This PR isn't related to the grenades count. It fixes bug which happens after flashbang was thrown.

**Comment by IgnacioFDM** [comment]
Yes, but if I have say, 3 smoke nades, and throw one, do they remain active? I would guess from @KaidoRen reply that they do.

**Comment by KaidoRen** [comment]
yes
smoke and he grenades remain active after throw

File: regamedll/dlls/game.cpp (hunk 2 above)
Comment (Dr-Strange91) [reply regamedll/dlls/game.cpp]: mp_flashbang_remain_active
Better :/

Comment (KaidoRen) [reply regamedll/dlls/game.cpp]: okey

**Comment by Freeman-AM** [comment]
What prevent this from beeing pulled ?

File: regamedll/dlls/wpn_shared/wpn_flashbang.cpp [hunk 3]
```diff
@@ -220,6 +220,16 @@ void CFlashbang::WeaponIdle()
...
+
+#ifdef REGAMEDLL_ADD
+		if (flashbang_remain_active.value > 0 && m_pPlayer->m_rgAmmo[m_iPrimaryAmmoType])
+		{
+			SendWeaponAnim(FLASHBANG_DRAW, UseDecrement() != FALSE);
+			m_flTimeWeaponIdle = UTIL_WeaponTimeBase() + RANDOM_FLOAT(10, 15);
+			m_flReleaseThrow = -1.0f;
+			return;
+		}
+#endif
```
Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/wpn_shared/wpn_flashbang.cpp]: Wow! This is easier to read than original valve's logic:
https://github.com/s1lentq/ReGameDLL_CS/blob/26db32e8845647cf024e38f11cb3da3071ac0dd1/regamedll/dlls/wpn_shared/wpn_hegrenade.cpp#L222-L236
We should refactor hegrenade and smokegrenade's code too.
But I prefer:
```C++
 // we've finished the throw, restart.
 m_flStartThrow = 0;
 
 if (m_pPlayer->m_rgAmmo[m_iPrimaryAmmoType] == 0)
 {
 	RetireWeapon();
 	return;
 }
 
 SendWeaponAnim(HEGRENADE_DRAW, UseDecrement() != FALSE);
 m_flTimeWeaponIdle = UTIL_WeaponTimeBase() + RANDOM_FLOAT(10, 15);
 m_flReleaseThrow = -1.0f;
```
So, for flashbang:
```C++
 // we've finished the throw, restart.
 m_flStartThrow = 0;

#ifdef REGAMEDLL_ADD
 if (m_pPlayer->m_rgAmmo[m_iPrimaryAmmoType] == 0)
 {
 	RetireWeapon();
 	return;
 }
 
 SendWeaponAnim(FLASHBANG_DRAW, UseDecrement() != FALSE);
 m_flTimeWeaponIdle = UTIL_WeaponTimeBase() + RANDOM_FLOAT(10, 15);
 m_flReleaseThrow = -1.0f;
#else // REGAMEDLL_ADD
 RetireWeapon();
#endif
```

**Comment by Vaqtincha** [comment]
зачем квар ? бред и полный провал.

**Comment by WPMGPRoSToTeMa** [comment]
@KaidoRen why did you close it?

**Pull request closed without merging**